tests/src/tests/da/api.rs

```diff
@@ -52,3 +56,62 @@ async fn test_get_blob_data() {
 
     assert!(blob_data.is_some());
 }
+
+#[tokio::test]
+async fn test_block_peer() {
+    let topology = Topology::spawn(TopologyConfig::validator_and_executor()).await;
+    let executor = &topology.executors()[0];
+
+    let blacklisted_peers = executor.blacklisted_peers().await;
+    assert!(blacklisted_peers.is_empty());
+
+    let membership = executor
+        .config()
+        .da_network
+        .backend
+        .validator_settings
+        .membership
+        .members();
+
+    // take second peer ID from the membership set
+    let existing_peer_id = membership
+        .iter()
+        .nth(1)
+        .expect("Expected at least 2 members in the set");
+
+    // try block/unblock peer id combinations
+    let blocked = executor.block_peer(existing_peer_id.to_string()).await;
+    assert!(blocked);
+
+    let blacklisted_peers = executor.blacklisted_peers().await;
+    assert_eq!(blacklisted_peers.len(), 1);
+    assert_eq!(blacklisted_peers[0], existing_peer_id.to_string());
+
+    let blocked = executor.block_peer(existing_peer_id.to_string()).await;
+    assert!(!blocked);
+
+    let unblocked = executor.unblock_peer(existing_peer_id.to_string()).await;
+    assert!(unblocked);
+
+    let blacklisted_peers = executor.blacklisted_peers().await;
+    assert!(blacklisted_peers.is_empty());
+
+    let unblocked = executor.unblock_peer(existing_peer_id.to_string()).await;
+    assert!(!unblocked);
+
+    // try blocking/unblocking non existing peer id
```

Comment:
These tests allow blocking/unblocking non existing peers. The support for peers (via membership) could be added in a different PR.